internal/server/v2/parser.go

```diff
@@ -45,3 +47,91 @@ func splitArc(s string) ([]string, error) {
 	}
 	return parts, nil
 }
+
+func parseArc(s string) (*Arc, error) {
+	if len(s) < 2 {
+		return nil, status.Errorf(
+			codes.InvalidArgument, "invalid arc string: %s", s)
+	}
+	arc := &Arc{}
+	if s[0:2] == "->" {
+		arc.out = true
+	} else if s[0:2] == "<-" {
+		arc.out = false
+	} else {
+		return nil, status.Errorf(
+			codes.InvalidArgument, "arc string should start with arrow, %s", s)
+	}
+	s = s[2:]
+	// No property defined; This is to fetch all the properties.
+	if len(s) == 0 {
+		return arc, nil
+	}
+	if s[0] == '[' {
+		if s[len(s)-1] != ']' {
+			return nil, status.Errorf(
+				codes.InvalidArgument, "invalid filter string: %s", s)
+		}
+		s = s[1 : len(s)-1]
+		// [prop1, prop2]
```

Comment:
At this point, [ and ] are already removed, so:   prop1, prop2 ?

Reply:
Moved the comment to the beginning of the block